src/tool.ts

```diff
@@ -23,6 +32,148 @@ export async function printResults(tool: string, sarifFile: string) {
   endGroup()
 }
 
+export function splitStringAtFirstSlash(inputString: string | undefined): [string, string] {
+  if (inputString != null) {
+    const [firstPart, secondPart] = inputString.split('/', 2)
+    return [firstPart, secondPart]
+  }
+  return ['', '']
+}
+
+export const options: Partial<SimpleGitOptions> = {
+  baseDir: process.cwd(),
+  binary: 'git',
+  maxConcurrentProcesses: 6,
+  trimmed: false,
+}
+
+export async function prForFixSuggestion(
+  jsonFile: string,
+  fixId: string,
+  repoOwner: string,
+  repoName: string
+) {
+  let newBranch: string = 'codesec/sca/'
+
+  // git configuration
+  const git = simpleGit(options)
+  await git.addConfig('user.name', 'CodeSec Bot', false, 'global')
+  await git.addConfig('user.email', 'codesec-eng@lacework.com', false, 'global')
+
+  // get current branch
+  // trigger: on pr
+  let currBranch = getOptionalEnvVariable('GITHUB_HEAD_REF', '')
+  if (currBranch == '') {
+    // trigger: on push
+    currBranch = getRequiredEnvVariable('GITHUB_REF_NAME')
+  }
+
+  newBranch += currBranch + '/'
+
+  var patchReport = 'patchSummary.md'
+
+  // create command to run on branch
+  var args = ['sca', 'patch', '.', '--sbom', jsonFile, '--fix-id', fixId, '-o', patchReport]
+
+  // call patch command
+  await callLaceworkCli(...args)
+
+  let patch = readFileSync(patchReport, 'utf-8')
+  // title is the first line of the patch summary
+  let titlePR = patch.split('\n')[0].substring(2)
+  newBranch += titlePR.split('bump ')[1].split(' to')[0].replaceAll(' ', '_').replaceAll(':', '-')
+  if (newBranch[newBranch.length - 1] == '.') {
+    newBranch = newBranch.substring(0, newBranch.length - 1)
+  }
+
+  // check if branch already exists for branch creation/overwrite
+  let branchList = (await git.branch()).all
+  let found = false
+  await git.fetch()
+  for (let branch of branchList) {
+    if (branch.includes(newBranch)) {
+      found = true
+      break
+    }
+  }
+
+  // create local branch
+  await git.checkoutLocalBranch(newBranch)
+
+  // parse the modified files from the patch summary
+  let files: string[] = []
+  let text: string = patch.split('## Files that have been modified:')[1]
+  if (text == undefined) {
+    return
+  }
+  let lines: string[] = text.split('-')
+  for (let line of lines) {
+    // delete whitespaces
+    line = line.trimStart().trimEnd()
+    // delete the '*'
+    line = line.substring(1, line.length - 1)
+    files.push(line)
+  }
+
+  // add modified files to branch
+  for (const file of files) {
+    if (file != '') {
+      await git.add(file)
+    }
+  }
+
+  // commit and push changes --force to overwrite remote branch
+  await git.commit('Fix for: ' + newBranch + '.').push('origin', newBranch, ['--force'])
+
+  // open PR:
+  if (!found) {
```

Comment:
I'm not sure if using whether the branch exists is the right way to determine whether we should open a PR or update an existing one. At the moment if the previous PR was closed but the branch was not deleted then we will essentially give up on trying to get the customer to update that dependency which isn't ideal. I think we should look for an open PR that matches the branch and update that one if we find it, and otherwise open a new one.